nevergrad/optimization/base.py

```diff
@@ -736,3 +749,25 @@ def __eq__(self, other: tp.Any) -> tp.Any:
             if self._config == other._config:
                 return True
         return False
+
+
+def _constraint_solver(parameter: p.Parameter, budget: int) -> p.Parameter:
+    """Runs a suboptimization to solve the parameter constraints"""
+    parameter_without_constraint = parameter.copy()
+    parameter_without_constraint._constraint_checkers.clear()
+    opt = registry["OnePlusOne"](parameter_without_constraint, num_workers=1, budget=budget)
+    opt._constraints_manager.max_trials = 1
+    for _ in range(budget):
+        cand = opt.ask()
+        # Our objective function is minimum for the point the closest to
+        # the original candidate under the constraints.
+        penalty = cand._constraint_penalties()
+        if not penalty > 0:  # constraints are satisfied
+            return cand
```

Comment:
don't return this, since this candidate does not have any constraint anymore listed, it could create weird bugs.

Reply:
oh my god thx.